synapse/storage/group_server.py

```diff
@@ -647,6 +672,60 @@ def is_user_invited_to_local_group(self, group_id, user_id):
             allow_none=True,
         )
 
+    def get_users_membership_info_in_group(self, group_id, user_id):
+        """Get a dict describing the membership of a user in a group.
+
+        Example if joined:
+
+            {
+                "membership": "joined",
+                "is_public": True,
+                "is_privileged": False,
+            }
+
+        Returns an empty dict if the user is not joined/invited/etc
+        """
+        def _get_users_membership_in_group_txn(txn):
+            row = self._simple_select_one_txn(
+                txn,
+                table="group_users",
+                keyvalues={
+                    "group_id": group_id,
+                    "user_id": user_id,
+                },
+                retcols=("is_admin", "is_public"),
+                allow_none=True,
+            )
+
+            if row:
+                return {
+                    "membership": "joined",
```

Comment:
Hmm, I thought we generally used "join" rather than "joined", "invite" rather than "invited" and "leave" rather than "left" etc.